src/core/Header/HeaderButton.tsx

```diff
@@ -0,0 +1,88 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import '@itwin/itwinui-css/css/header.css';
+
+import cx from 'classnames';
+import React from 'react';
+import {
+  Button,
+  ButtonProps,
+  DropdownButton,
+  DropdownButtonProps,
+} from '../Buttons';
+
+import { useTheme } from '../utils/hooks/useTheme';
+
+export type HeaderButtonProps = {
+  /**
+   * Name.
+   */
+  name: React.ReactNode;
+  /**
+   * Description
+   */
+  description?: React.ReactNode;
+  /**
+   * Is the menu active.
+   * @default false
+   */
+  isActive?: boolean;
+} & Omit<
+  Partial<DropdownButtonProps> & Partial<ButtonProps>,
+  'children' | 'styleType'
+>;
+
+/**
+ * Header button that handles slim state of the `Header` it's in.
+ * When in slim mode, will only display the name and reduce icon size.
+ * Will display an arrow on the right if `menuItems` prop is provided.
+ * @example
+ * <HeaderButton name='Project A' description='0n00434' />
+ * <HeaderButton name='Project B' description='0n00434' startIcon={<SvgImodelHollow />}/>
+ * <HeaderButton name='Project C' startIcon={<img src='project.png' />}/>
+ * <HeaderButton name='Project D' isActive />
+ * <HeaderButton name='Project D' isActive menuItems={...}/>
```

Comment:
Minor but can you make this one as the first example as this one should be used the most.

Reply:
Also, if you could add space before closing tags, where missed... :D

Reply:
Moved the menuItems part to the Project A example